src/main/kotlin/com/demonwav/mcdev/util/SemanticVersion.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.util
+
+import com.demonwav.mcdev.util.SemanticVersion.Companion.VersionPart.PreReleasePart
+import com.demonwav.mcdev.util.SemanticVersion.Companion.VersionPart.ReleasePart
+
+class SemanticVersion(val value: String) : Comparable<SemanticVersion> {
+    val parts = value.split(".").map {
+        if (it.contains("_pre")) {
+            val subParts = it.split("_pre")
+            PreReleasePart(subParts[0].toInt(), subParts[1].toInt())
```

Comment:
In the context of MC versions, we can be 100% confident, I think, but I'll add some safeguards to prevent more general purpose usage blowing up.